src/main/java/javaslang/collection/Tree.java

```diff
@@ -247,7 +247,7 @@ String toString(Tree<T> tree, int depth) {
                         final String children = tree.getChildren()
                                 .map(child -> toString(child, depth + 1))
                                 .join();
-                        return String.format("\n%s%s%s", indent, value, children);
+                        return String.format("%n%s%s%s", indent, value, children);
```

Comment:
`String#format()` is rather slow operation.
Since there is no special formatting I'd recommend to use normal String concatenation or using StringBuilder. It is faster and still readable.